controllers/storagecluster/storageclasses.go

```diff
@@ -114,6 +114,22 @@ func (r *StorageClusterReconciler) createStorageClasses(sccs []StorageClassConfi
 				skippedSC = append(skippedSC, sc.Name)
 				continue
 			}
+			// if rados namespace is provided, update the `sc cluster-id = rados-namespace cluster-id``
+			if strings.Contains(sc.Name, "-rados-namespace") {
+				radosNamespace := cephv1.CephBlockPoolRadosNamespace{}
+				key = types.NamespacedName{Name: radosNamspaceName, Namespace: sc.Parameters["clusterID"]}
+				err := r.Client.Get(context.TODO(), key, &radosNamespace)
+				if err != nil || radosNamespace.Status == nil || radosNamespace.Status.Phase != cephv1.ConditionType(util.PhaseReady) || radosNamespace.Status.Info["clusterID"] == "" {
+					r.Log.Info("Waiting for radosNamespace to be Ready. Skip reconciling StorageClass",
+						"radosNamespace", klog.KRef(key.Namespace, key.Name),
+						"it's Status %+v", radosNamespace.Status,
```

Comment:
i dont think we log complete Status in log

Reply:
It's need to know why it fails

Reply:
It's always available in the CR and we collect it separately. Let's not log this.

Reply:
Updated